### ISSUE TITLE
Use SysAllocStringLen instead of SysAllocString to allocate BSTR's

### ISSUE THREAD

**Issue by mgspross** [go-ole/go-ole#29]
The library currently uses `SysAllocString` to allocate BSTR's when strings are passed to COM functions. I'd recommend using `SysAllocStringLen` instead, due to limitations with `SysAllocString`.

The problem with the current implementation is that a BSTR is actually allowed to contain binary data, including `NUL` characters. `SysAllocString` doesn't support this contract (it terminates the string when it finds a `NUL` character), but `SysAllocStringLen` does.

In other words, `SysAllocStringLen` lets you pass a pointer to a string, plus the length of the string, which allows you allocate strings that contain arbitrary bytes. I think this also makes sense in the context of Go as well, since strings in Go can also hold arbitrary bytes.

This change also won't change functionality when passing "normal" strings around, so it should be low-risk.
